stackslib/src/config/mod.rs

```diff
@@ -1210,53 +1210,321 @@ impl std::default::Default for Config {
+    ///
+    /// Default: `"mocknet"`
     pub mode: String,
+    /// The network-specific identifier used in P2P communication and database initialization .
+    /// Derived from `mode` during config load, unless explicitly overridden (for test purposes).
+    ///
+    /// Default: Derived from [`BurnchainConfig::mode`] ([`CHAIN_ID_MAINNET`] for `mainnet`, [`CHAIN_ID_TESTNET`] otherwise).
     pub chain_id: u32,
+    /// The peer protocol version number used in P2P communication.
+    /// This parameter cannot be set via the configuration file.
+    ///
+    /// Default: Derived from [`BurnchainConfig::mode`] ([`PEER_VERSION_MAINNET`] for `mainnet`, [`PEER_VERSION_TESTNET`] otherwise).
     pub peer_version: u32,
+    /// Specifies a mandatory wait period (in milliseconds) after receiving a burnchain tip
+    /// before the node attempts to build the anchored block for the new tenure.
+    /// This duration effectively schedules the start of the block-building
+    /// process relative to the tip's arrival time.
+    /// Only when [`BurnchainConfig::mode`] is `"helium"` or `"mocknet"`.
+    ///
+    /// Default: `5000` milliseconds.
+    ///          `10000` milliseconds when launched with the `helium` or `mocknet` subcommands.
     pub commit_anchor_block_within: u64,
+    /// The Maximum amount (in sats) of "burn commitment" to broadcast for the next block's leader election.
+    ///  Acts as a safety cap to limit the maximum amount spent on mining.
+    ///
+    /// It serves as both the target fee and a fallback if dynamic fee calculations fail or cannot be performed
+    ///
+    /// This setting can be hot-reloaded from a miner's config file, allowing adjustment without restarting.
+    /// Primarily relevant for miners.
+    ///
+    /// Default: `20000` satoshis
     pub burn_fee_cap: u64,
+    /// The hostname or IP address of the bitcoin node peer.
+    ///
+    /// This field is required for all node configurations as it specifies where to find the underlying
+    /// bitcoin node to interact with for PoX operations, block validation, and mining.
+    ///
+    /// Default: `"0.0.0.0"`
     pub peer_host: String,
+    /// The P2P network port of the bitcoin node specified by [`BurnchainConfig::peer_host`].
+    ///
+    /// Default: `8333`
     pub peer_port: u16,
+    /// The RPC port of the bitcoin node specified by [`BurnchainConfig::peer_host`].
+    ///
+    /// Default: `8332`
     pub rpc_port: u16,
+    /// Flag indicating whether to use SSL/TLS when connecting to the bitcoin node's RPC interface.
+    ///
+    /// Default: `false`
     pub rpc_ssl: bool,
+    /// The username for authenticating with the bitcoin node's RPC interface.
+    /// Required if the bitcoin node requires RPC authentication.
+    ///
+    /// Default: `None`
     pub username: Option<String>,
+    /// The password for authenticating with the bitcoin node's RPC interface.
+    /// Required if the bitcoin node requires RPC authentication.
```

Comment:
This seems inconsistent, otherwise we would want to add `Required if...` to several other settings in this PR. 
i.e.
```
Required if bitcoin p2p is on a non-default port
```
I'm not convinced of either option, it just seems a bit inconsistent is all.